packages/core/src/Modules/Cashier/Containers/cashier.jsx

```diff
@@ -0,0 +1,119 @@
+import PropTypes         from 'prop-types';
+import React             from 'react';
+import { withRouter }    from 'react-router-dom';
+import { localize }      from 'App/i18n';
+import { FadeWrapper }   from 'App/Components/Animations';
+import VerticalTab       from 'App/Components/Elements/VerticalTabs/vertical-tab.jsx';
+import routes            from 'Constants/routes';
+import { connect }       from 'Stores/connect';
+import WalletInformation from '../../Reports/Containers/wallet-information.jsx';
+
+class Cashier extends React.Component {
+    setWrapperRef = (node) => {
+        this.wrapper_ref = node;
+    };
+
+    handleClickOutside = (event) => {
+        if (this.wrapper_ref && !this.wrapper_ref.contains(event.target)) {
+            this.props.history.push(routes.trade);
+        }
+    };
+
+    componentDidMount() {
+        this.props.enableRouteMode();
+        document.addEventListener('mousedown', this.handleClickOutside);
+        this.props.toggleCashier();
+        // we still need to populate the tabs shown on cashier
+        this.props.onMount();
+    }
+
+    componentWillUnmount() {
+        this.props.toggleCashier();
+        this.props.disableRouteMode();
+        document.removeEventListener('mousedown', this.handleClickOutside);
+    }
+
+    render () {
+        const menu_options = () => {
+            const options = [];
+
+            this.props.routes.forEach(route => {
+                if ((route.path !== routes.cashier_pa || this.props.is_payment_agent_visible) &&
+                    (route.path !== routes.cashier_pa_transfer || this.props.is_payment_agent_transfer_visible)) {
+                    options.push({
+                        default: route.default,
+                        icon   : route.icon_component,
+                        label  : route.title,
+                        value  : route.component,
+                        path   : route.path,
+                    });
+                }
+            });
+
+            return options;
+        };
+
+        const action_bar_items = [
+            {
+                onClick: () => {
+                    this.props.history.push(routes.trade);
+                    this.props.toggleCashier();
+                },
+                icon : 'ModalIconClose',
+                title: localize('Close'),
+            },
+            {
+                component: () => <WalletInformation />,
+                title    : '',
+            },
+        ];
+        return (
+            <FadeWrapper
```

Comment:
this means that everytime the routes changes, there will be fade animation . for the whole vertical tab?

Reply:
it only happens on mount and unmount of the whole component. I'm following the same way that reports page is doing it.